DistroLauncher/ExitStatus.cpp

```diff
@@ -0,0 +1,203 @@
+
+        // Polls WSL to ensure the distro is actually stopped.
+        bool ensureDistroStopped(unsigned int maxNoOfRetries)
+        {
+            for (unsigned int i = 0; i < maxNoOfRetries; ++i) {
+                auto runner = Helpers::ProcessRunner(L"wsl -l --quiet --running");
+                auto exitCode = runner.run();
+                if (exitCode != 0L) {
+                    return false;
+                }
+
+                // Returns true once we don't find the DistributionInfo::Name in process's stdout
+                auto output = runner.getStdOut();
+                if (output.find(DistributionInfo::Name) == std::wstring::npos) {
+                    return true;
+                }
+
+                // We don't need to be hard real time precise.
+                Sleep(997); // NOLINT(readability-magic-numbers): only used here.
+            }
+            return false;
+        }
+
+        VoidResult act(const KeyValuePairs& launcherCmds)
+        {
+            auto actionIt = launcherCmds.find("action");
+            if (actionIt == launcherCmds.end()) {
+                // nothing to do.
+                return VoidResult();
+            }
+
+            // I'm hardcoding that "action" expects strings.
+            auto action = std::any_cast<std::string>(actionIt->second);
+
+            auto f = capabilities.find(action);
+            if (f == capabilities.end()) {
+                return nonstd::make_unexpected(std::domain_error(action));
+            }
+
+            auto res = f->second();
+            if (!res.has_value()) {
+                return nonstd::make_unexpected(res.error());
+            }
+
+            return VoidResult();
+        }
+
+        // Since there is only one kind configuration entry expected at this moment, it would not make a lot of sense to
+        // create enums or tables to work with that.
+        VoidResult config(const KeyValuePairs& launcherCmds)
+        {
+            auto configIt = launcherCmds.find("defaultUid");
+            if (configIt == launcherCmds.end()) {
+                // nothing to do
+                return VoidResult();
+            }
+            auto default_uid = std::any_cast<unsigned long>(configIt->second);
+            // TODO: Replace the default flags on this call by the current ones reading before writting.
+            // The planned wrapper class should resolve this.
+            auto hr = g_wslApi.WslConfigureDistribution(default_uid, WSL_DISTRIBUTION_FLAGS_DEFAULT);
+            if (FAILED(hr)) {
+                return nonstd::make_unexpected(std::runtime_error(
+                  "Could not configure distro to the new default UID: " + std::to_string(default_uid)));
+            }
+
+            return VoidResult();
+        }
+
+        namespace Actions
+        {
+            VoidResult shutdownDistro()
+            {
+                const std::wstring shutdownCmd = L"wsl -t " + DistributionInfo::Name;
```

Comment:
is there a reason to allocate an std::wstring just to get an std::string below?

Reply:
`shutdownCmd.c_str()` returns a `const wchar_t*`, which is what the `_wsystem()` function expects. The wstring allocation was to easily concatenate the command `wsl -t`with the `DistributionInfo::Name`. That could be done concatenating C-strings, it would be pretty much the same, just a little bit more complicated.

Reply:
Ok, that's fine. I've been totally spoiled by QString